x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/transport/SecurityServerTransportInterceptor.java

```diff
@@ -386,6 +386,16 @@ private <T extends TransportResponse> void sendWithRemoteAccessHeaders(
                         remoteClusterAlias,
                         authentication.getEffectiveSubject(),
                         ActionListener.wrap(roleDescriptorsIntersection -> {
+                            if (roleDescriptorsIntersection.isEmpty()) {
```

Comment:
My preference is to check and fail in the interceptor. Its behaviour is consistent with the existing CCS behaviour and seems to be more reasonable. 

Failure at here let each individual action decide how they want to deal with the exception. For search, it decides to fail the overall request. But that is particular to search. Other actions can choose different behaviours. For example, FieldCaps choose to embed the failure in a `failure` section in the response which contains successful respones for other indices. That is, the request `GET *:*:*/_field_caps?fields=*` will not fail entirely as long as the user has access to some local indices. It also allows each remote cluster to fail independently. Another example is ResolveIndex which choose to ignore any failure and just return empty. If we check and fail at Authorization time, it will have to fail the entire request unless we implement something like how bulk request is handled. It is a lot of work and rather little gain.

There is one future improvement we could make in the Authorization area. Right now, `*:*` is resolved to all remote clusters. A user may only have access to some of them, but we send requests to all of them anyway. For search, this means the entire request is going to fail. This is rather surprising because a remote cluster can be defined without the user knowing it and it ends up breaking existing queries for the user. Legacy remote cluster has to live with this because there is no way to define what remote clusters a user has access. With the new RCS, we now have a way to resolve target clusters in the user's context to avoid this issue.